### PR TITLE
arrays-programs_involving_data_sequences-hard-q10

### PR DESCRIPTION
**Pull request by te-sa**
## PR Checklist
- [x]  Are helper methods really needed here?
    > Opt for hardcoding input/output into the input provider when possible
- [x]  Are the assertion fail messages good?
    - [ ]  Logical (does it make sense?)
    - [x]  Readable (can a student understand what’s wrong?)
    - [x]  Not too feed-y (does it give away the answer?)
    - [x]  Grammatically sound (is the grammar and punctuation in the sentence proper?)
    - [x]  Specific (does the fail message relate to the question?)
- [x]  Is the code formatted well?
    > Run `CTRL` + `ALT` + `L` in IntelliJ
- [x]  EOF newlines
- [ ]  Do the comments for Parsons follow the correct convention?
- [ ]  Is the question being tested sufficiently?
    - [ ]  Are there representative cases to test the program?
    - [ ]  What about edge cases? (empty strings, negatives, infinity, off-by-one errors, etc)
        > If the question doesn’t specify what to do, mark it as a bad question
- [x]  Are the test method names good?
    > Test method names show up in a “What went well” section on the website, so “mathTest” will become “What went well: math test”, which doesn’t make much sense. A better name in this example would be “calculatesTaxCorrectly” so it shows up as “What went well: calculates tax correctly”

### CODE REVIEW

File: src/arrays/programs_involving_data_sequences/hard/q10/MainTest.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    @Override
+    public void runMain() {
+        ArrayThereDelilah.main(new String[0]);
+    }
+
+    static Stream<Arguments> inputProvider() {
+        return Stream.of(
+                Arguments.of(new String[]{"add", "here", "lilly", "president", "jelly", "advance", "archers"}, "delilah".toCharArray()),
+                Arguments.of(new String[]{"archers", "get", "hello", "island", "cloud"}, "hello".toCharArray()),
+                Arguments.of(new String[]{"fireworks", "acorn", "array", "pilot", "hedge"}, "world".toCharArray()),
```
Comment (Hedgemon4) [inline src/arrays/programs_involving_data_sequences/hard/q10/MainTest.java:36]: You can also take a look at arrayUtil for this question. There is a method that will generate an array of random words for you, which would be useful for this method. However, if you did that you would have to create a helper method, but that should be fine as long as you don't look at the original code.

**Review by Hedgemon4 — CHANGES_REQUESTED** [review]

File: src/arrays/programs_involving_data_sequences/hard/q10/MainTest.java (hunk 1 above)
Comment (te-sa) [reply src/arrays/programs_involving_data_sequences/hard/q10/MainTest.java:38]: I made that change but also kept the hardcoded ones.

File: src/arrays/programs_involving_data_sequences/hard/q10/MainTest.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    void correctMiddleMethod(String[] input, char[] word) throws Throwable {
+        Object[][] arguments = {
+                {input, String[].class},
+        };
+        MethodTest m = new MethodTest(ArrayThereDelilah.class, "middle", arguments);
+        Object output = m.callMethod();
+        CustomAssertions._assertArrayEquals(word, output, "Your middle method does not return the correct array of middle characters.");
```
Comment (Hedgemon4) [inline src/arrays/programs_involving_data_sequences/hard/q10/MainTest.java:67]:
```suggestion
        CustomAssertions._assertArrayEquals(word, output, "Your middle method does not a character array with the middle character of each word.");
```

File: src/arrays/programs_involving_data_sequences/hard/q10/ArrayThereDelilah.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
+package arrays.programs_involving_data_sequences.hard.q10;
+
+/*
+Create a Java program that takes an array of strings, finds the middle character in each string, and prints it to the
+console. To do this, create a method called middle, which accepts a String array, finds the middle character in the
+string, and adds it to a character array. For this method, find the middle element by dividing the length by two,
+rounding the number, and subtracting one. After computing the array, return it and print each character in the main
+method. Use the provided main method to call your method and print the output.
```
Comment (Hedgemon4) [inline src/arrays/programs_involving_data_sequences/hard/q10/ArrayThereDelilah.java:8]:
```suggestion
method. Use the provided main method to call your method and print the output. Note that your middle method should be general enough to work with input of any length.
```

**Review by Hedgemon4 — CHANGES_REQUESTED** [review]

**Review by Hedgemon4 — APPROVED** [review]

**Pull request merged**